test/public.spec.js

```diff
@@ -0,0 +1,423 @@
+
+        expect(textLines).toContain('Display: 151 x 71');
+        expect(textLines).toContain('Natural: 100 x 80');
+        expect(textLines).toContain('Image coverage: 74.62%');
+        expect(textLines).toContain('File Size: 4.464 KB');
+      });
+
+      it('should click to new tab', function() {
+        let urlLink = getImageOverlay(0);
+
+        expect(urlLink.href).toEqual(jasmine.stringMatching('test/assets/placeholder-100x80.png'));
+        expect(urlLink.target).toEqual('_blank');
+      });
+    });
+
+    describe('small images overlays', function() {
+      beforeEach(function() {
+        createDomNodes([
+          createSmallImg()
+        ]);
+        window.NCC.imageChecker.showImagesInfo();
+      });
+
+      it('should only have title', function() {
+        let imageOverlay = getImageOverlay(0);
+        let titleParts = imageOverlay.title.split(',');
+        expect(titleParts[0].trim()).toEqual('Coverage: 106.67%');
+        expect(titleParts[1].trim()).toEqual('File Size: 4.464 KB');
+        expect(titleParts[2].trim()).toEqual(jasmine.stringMatching('URL: .*test/assets/placeholder-100x80.png'));
+      });
+
+      it('should not have any content', function() {
+        let textLines = getImageOverlayTextLines(0);
+        expect(textLines).toEqual(['']);
+      });
+    });
+
+    describe('cross domain images overlays', function() {
+      beforeEach(function() {
+
+      });
+
+      it('should not show file size', function(done) {
+        createDomNodes([
+          createCrossDomainImg()
+        ]);
+        spyOn(window.performance, 'getEntriesByName').and.returnValue([{encodedBodySize: 0}]);
+        crossDomainImg.onload = function() {
+          window.NCC.imageChecker.showImagesInfo();
+
+          let textLines = getImageOverlayTextLines(0);
+          expect(textLines).toContain('File size unavailable');
+          done();
+        };
+      });
+    });
+  });
+
+  describe('hideImagesInfo()', function() {
+    it('should remove all image overlays', function() {
+      createDomNodes([
+        createBackgroundImg(),
+        createNoBackgroundImg(),
+        createHiddenImg(),
+        createInvisibleImg(),
+        createNoSrcImg(),
+        createNoDimensionImg(),
+        createSmallImg(),
+        createMediumImg(),
+        createBigImg(),
+        createImg()
+      ]);
+      window.NCC.imageChecker.showImagesInfo();
+
+      window.NCC.imageChecker.hideImagesInfo();
+
+      let siblings = getAllSiblingsAfter(images);
+      expect(siblings.length).toEqual(0);
+    });
+  });
+
+  describe('isImagesInfoActive()', function() {
+    it('should return true', function() {
+      createDomNodes([
+        createImg()
+      ]);
+      window.NCC.imageChecker.showImagesInfo();
+
+      expect(window.NCC.imageChecker.isImagesInfoActive()).toEqual(true);
+    });
+
+    it('should return false', function() {
+      createDomNodes();
+      window.NCC.imageChecker.showImagesInfo();
```

Comment:
@tabanliviu COVERAGE - This is testing more than the scenario's description as the test has knowledge that no active overlays means the info won't show.